Source/Dafny/Dafny.atg

```diff
@@ -2202,8 +2203,8 @@ UpdateStmt<out Statement/*!*/ s>
       ]
       Expression<out suchThat, false, true>
     | ":-"                         (. x = t; .)
-      [ IF(la.kind == _expect)     /* an Expression can also begin with an "expect", so this says to resolve it to pick up any "expect" here */
-        "expect"                   (. exceptionExpect = t; .)
+      [ IF(la.kind == _expect || la.kind == _assert || la.kind == _assume)     /* an Expression can also begin with an "expect", so this says to resolve it to pick up any "expect" here */
```

Comment:
I suggest making some `Is...` method that returns true for these 3 tokens. Then, it can also be used on L2217 and L2345.

Also, the comment (here and on L2217 and L2345) also is true of `assert` and `assume`.

Reply:
Done